backend/install.sh

```diff
@@ -6,7 +6,39 @@
 
 # Prerequisites
 
-sudo apt install -y python3-venv default-jre
+# Only try to install dependencies if we actually need them.
+
+PKG_FILE=/tmp/installed-package-list.txt
+DEPENDENCIES="python3-venv default-jre"
+TO_INSTALL=""
+
+dpkg -l | grep "^ii" > $PKG_FILE
+
+for PACKAGE in "${DEPENDENCIES[@]}"
+do
+    grep -w "ii $PACKAGE " $PKG_FILE > /dev/null
+    if [[ $? -ne 0 ]]
+    then
+        TO_INSTALL=$TO_INSTALL" "$PACKAGE
+    fi
+done
+
+if [ "$TO_INSTALL" != "" ]
+then
+    sudo apt-get update > /dev/null
+    if [ $? -ne 0 ] 
```

Comment:
I'd change this to:

```shell
    if sudo apt-get -qq update
```

This will allow you to add a `set -e` earlier, so that errors don't pass silently. All shell scripts should have that IMO — as an example, otherwise, the install might fail and no-one'd notice. I saw that in the old installer for `mox`.